utils.go

```diff
@@ -66,26 +68,44 @@ func getQuerySnippet(req *http.Request) string {
 
 	u := getDecompressor(req)
 	if u == nil {
-		return data
+		if len(query) != 0 && len(data) != 0 {
+			query += "\n"
+		}
+
+		return query + data
 	}
 	bs := bytes.NewBufferString(data)
 	b, err := u.decompress(bs)
 	if err == nil {
-		return string(b)
+		if len(query) != 0 && len(b) != 0 {
+			query += "\n"
+		}
+
+		return query + string(b)
 	}
 	// It is better to return partially decompressed data instead of an empty string.
 	if len(b) > 0 {
-		return string(b)
+		if len(query) != 0 && len(b) != 0 {
+			query += "\n"
+		}
+
+		return query + string(b)
 	}
+
 	// The data failed to be decompressed. Return compressed data
 	// instead of an empty string.
-	return data
+	if len(query) != 0 && len(data) != 0 {
+		query += "\n"
+	}
+
+	return query + data
 }
 
 // getFullQuery returns full query from req.
 func getFullQuery(req *http.Request) ([]byte, error) {
```

Comment:
nit:
```
func getFullQuery(req *http.Request) ([]byte, error) {
	var result bytes.Buffer
	if q := req.URL.Query().Get("query"); q != "" {
		result.WriteString(q)
	}
	data, err := ioutil.ReadAll(req.Body)
	if err != nil {
		return nil, err
	}
	// restore body for further reading
	req.Body = ioutil.NopCloser(bytes.NewBuffer(data))
	if u := getDecompressor(req); u != nil {
		br := bytes.NewReader(data)
		b, err := u.decompress(br)
		if err != nil {
			return nil, fmt.Errorf("cannot uncompress query: %s", err)
		}
		data = b
	}
	
	if result.Len() != 0 && len(data) != 0 {
		result.WriteByte('\n')
	}
	result.Write(data)

	return result.Bytes(), nil
}
```

Reply:
I rewrite this code the same like getQuerySnippet from your comment below